extension-admob/src/admob_android.cpp

```diff
@@ -114,33 +119,53 @@ static void InitJNIMethods(JNIEnv* env, jclass cls)
     g_admob.m_ShowAdInspector = env->GetMethodID(cls, "showAdInspector", "()V");
     g_admob.m_UpdateBannerLayout= env->GetMethodID(cls, "updateBannerLayout", "()V");
 
+    g_admob.m_IsAppOpenLoaded = env->GetMethodID(cls, "isAppOpenLoaded", "()Z");
     g_admob.m_IsRewardedLoaded = env->GetMethodID(cls, "isRewardedLoaded", "()Z");
     g_admob.m_IsInterstitialLoaded = env->GetMethodID(cls, "isInterstitialLoaded", "()Z");
     g_admob.m_IsRewardedInterstitialLoaded = env->GetMethodID(cls, "isRewardedInterstitialLoaded", "()Z");
     g_admob.m_IsBannerLoaded = env->GetMethodID(cls, "isBannerLoaded", "()Z");
     g_admob.m_SetMaxAdContentRating = env->GetMethodID(cls, "setMaxAdContentRating", "(I)V");
 }
 
-void Initialize_Ext()
+void Initialize_Ext(dmExtension::Params* params, const char* defoldUserAgent)
 {
     dmAndroid::ThreadAttacher threadAttacher;
     JNIEnv* env = threadAttacher.GetEnv();
     jclass cls = dmAndroid::LoadClass(env, "com.defold.admob.AdmobJNI");
 
     InitJNIMethods(env, cls);
 
-    jmethodID jni_constructor = env->GetMethodID(cls, "<init>", "(Landroid/app/Activity;)V");
-
-    g_admob.m_AdmobJNI = env->NewGlobalRef(env->NewObject(cls, jni_constructor, threadAttacher.GetActivity()->clazz));
+    const char* appOpenAdUnitId = dmConfigFile::GetString(params->m_ConfigFile, "admob.app_open_android", 0);
+    jstring jappOpenAdUnitId = env->NewStringUTF(appOpenAdUnitId);
+    jstring jdefoldUserAgent = env->NewStringUTF(defoldUserAgent);
+    jmethodID jni_constructor = env->GetMethodID(cls, "<init>", "(Landroid/app/Activity;Ljava/lang/String;Ljava/lang/String;)V");
+    g_admob.m_AdmobJNI = env->NewGlobalRef(env->NewObject(cls, jni_constructor, threadAttacher.GetActivity()->clazz, jappOpenAdUnitId, jdefoldUserAgent));
+    env->DeleteLocalRef(jappOpenAdUnitId);
+    env->DeleteLocalRef(jdefoldUserAgent);
 }
 
 void Finalize_Ext()
 {
 }
 
-void Initialize(const char* defoldUserAgent)
+void Initialize()
+{
+    CallVoidMethodChar(g_admob.m_AdmobJNI, g_admob.m_Initialize);
```

Comment:
/extension-admob/src/admob_android.cpp
	Line 153: no matching function for call to 'CallVoidMethodChar'
    CallVoidMethodChar(g_admob.m_AdmobJNI, g_admob.m_Initialize);
    ^~~~~~~~~~~~~~~~~~

Reply:
did you mean `CallVoidMethod()`?

Reply:
Sigh, never do last minute changes without testing them....